packages/consensus/src/codec.cairo

```diff
@@ -31,15 +32,23 @@ pub impl EncodeSpan<T, +Encode<T>> of Encode<Span<T>> {
 }
 
 /// `Encode` trait implementation for `ByteArray`.
-/// TODO: use WordArray for arguments instead of ByteArray.
-/// Extra optimization: predict word array offset
 pub impl EncodeByteArray of Encode<ByteArray> {
     fn encode_to(self: @ByteArray, ref dest: WordArray) {
         encode_compact_size(self.len(), ref dest);
-        let num_bytes = self.len();
-        for i in 0..num_bytes {
-            dest.append_u8(self[i]);
-        }
+
+        // Serialized ByteArray: [num_bytes31_chunks, bytes31_chunks..., last_word, last_word_len]
```

Comment:
I would add a warning here, this code depends on the way `ByteArray` is serialized which is not enforced anywhere outside the `ByteArray`.

Reply:
Added. Yeah, I guess a better approach would be adding an iterator (over felts or even u32) to the ByteArray in corelib (related https://github.com/starkware-libs/cairo/issues/6817)